### PR TITLE
[JENKINS-49083] - Make the plugin compatible with Jenkins 2.102+

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-49083

- [x] - Make the plugin runnable with PCT by small pom.xml changes
- [x] - Fix the issue when a non-whitelisted class get serialized to the disk and loaded from there

@reviewbybees @jglick

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Build the plugin using https://github.com/jenkins-infra/pipeline-library
+buildPlugin()
```
Comment (jglick) [inline Jenkinsfile]: include 2.102 to prove the fix

**Comment by rinrinne** [comment]
Thx for this PR. I can merge this if test is OK on your local environment.

**Comment by oleg-nenashev** [comment]
@rinrinne I have no environment for that. Asked the issue reporter to test it (see https://issues.jenkins-ci.org/browse/JENKINS-49083)

P.S: There are still Javadoc issues

**Comment by rinrinne** [comment]
> @rinrinne I have no environment for that.
Aren't you running "mvn test" on your PR?? How did you verify your added test case? Compile only?

> P.S: There are still Javadoc issues
Unfortunately I have never seen javadoc issues because I'm compiling this plugin with javadoc in JDK7. The reason why you or CI are seeing javadoc issue is that doclint feature is enabled by default in javadoc for JDK8.

**Comment by rinrinne** [comment]
Hmm, I built this with JDK8, but no javadoc issues found.

```
$ mvn --batch-mode --errors --update-snapshots -Dmaven.test.failure.ignore clean install
(snip)
[INFO] --- maven-enforcer-plugin:3.0.0-M1:display-info (display-info) @ rabbitmq-consumer ---
[INFO] Maven Version: 3.5.2
[INFO] JDK Version: 1.8.0_151 normalized as: 1.8.0-151
[INFO] OS Info: Arch: amd64 Family: unix Name: linux Version: 4.10.0-42-generic
(snip)
[INFO] <<< maven-javadoc-plugin:2.10.4:javadoc (default) < generate-sources @ rabbitmq-consumer <<<
[INFO] 
[INFO] 
[INFO] --- maven-javadoc-plugin:2.10.4:javadoc (default) @ rabbitmq-consumer ---
[INFO] 
```
No errors or warnings were produced... I checked my local settings.xml but no related configurations.

How to reproduce javadoc issue locally?

**Comment by rinrinne** [comment]
Identified issue. I switched JDK from bundled OpenJDK 8 to Oracle JDK 8 Then build again. Issue can be reproduced.

```
$ mvn --batch-mode --errors --update-snapshots -Dmaven.test.failure.ignore clean install
(snip)
[INFO] --- maven-enforcer-plugin:3.0.0-M1:display-info (display-info) @ rabbitmq-consumer ---
[INFO] Maven Version: 3.5.2
[INFO] JDK Version: 1.8.0_161 normalized as: 1.8.0-161
[INFO] OS Info: Arch: amd64 Family: unix Name: linux Version: 4.10.0-42-generic
[INFO] 
(snip)
[INFO] <<< maven-javadoc-plugin:2.10.4:javadoc (default) < generate-sources @ rabbitmq-consumer <<<
[INFO] 
[INFO] 
[INFO] --- maven-javadoc-plugin:2.10.4:javadoc (default) @ rabbitmq-consumer ---
[INFO] 
2 errors
4 warnings
```

Seems javadoc's doclint feature is enabled by default in Oracle JDK8, but disabled in distribution bundled OpenJDK 8. I'm using Ubuntu 16.04.

Has this kind of Java environment issue already been shared within Jenkins community??

**Comment by rinrinne** [comment]
Anyway, I could pass added unit test with OpenJDK8. So I will merge this. Javadoc issue will be fixed later.
(disable doclint explicitly as short term solution)

**Pull request merged**